examples/python/6_schedule.py

```diff
@@ -0,0 +1,512 @@
+    def prefetch_schedule():
+        # Get nodes to be manipulated in the schedule.
+        k_loop = tkw.get_node_by_tag("k_loop")
+        load_a = tkw.get_node_by_tag_and_type("read_a", tkw.Read)
+        global_load_a, shared_load_a = tkw.partition_by_address_space(
+            load_a, GLOBAL_ADDRESS_SPACE
+        )
+        shared_write_a = tkw.get_node_by_tag_and_type("read_a", tkw.Write)
+        load_b = tkw.get_node_by_tag_and_type("read_b", tkw.Read)
+        global_load_b, shared_load_b = tkw.partition_by_address_space(
+            load_b, GLOBAL_ADDRESS_SPACE
+        )
+        shared_write_b = tkw.get_node_by_tag_and_type("read_b", tkw.Write)
+        mma = tkw.get_node_by_tag("mma")
+
+        # Create a pipeline with 2 stages and specify the operations that are overlapping.
+        with tkw.pipeline(k_loop) as pipelined_loop:
+            pipelined_loop.set_stage(
+                [
+                    (global_load_a, global_load_b),
+                    (shared_write_a, shared_write_b),
+                ],
+            )
+            pipelined_loop.set_stage(
+                [
+                    (shared_load_a, shared_load_b),
+                    (mma,),
+                ],
+            )
+
+    # Define compile options
+    M_val, N_val, K_val = shape
+    options = WaveCompileOptions(
+        subs={
+            M: M_val,
+            N: N_val,
+            K: K_val,
+            BLOCK_M: 128,
+            BLOCK_N: 256,
+            BLOCK_K: 64,
+            ADDRESS_SPACE: SHARED_ADDRESS_SPACE,
+            ADDRESS_SPACE_0: GLOBAL_ADDRESS_SPACE,
+            READ_SHARED_DELAY: 1,
+            WRITE_SHARED_DELAY: 1,
+            READ_GLOBAL_DELAY: 2,
+            WRITE_GLOBAL_DELAY: 2,
+            MMA_DELAY: 1,
+            VALU_DELAY: 1,
+            SHUFFLE_DELAY: 1,
+            SHARED_MEMORY_UNITS: 4,
+            GLOBAL_MEMORY_UNITS: 4,
+            MMA_UNITS: 4,
+            VALU_UNITS: 8,
+            SHUFFLE_UNITS: 8,
+        },
+        canonicalize=True,
+        schedule=SchedulingType.MANUAL,
+        print_ir_after="all" if is_debug else [],
+    )
+
+    # Set runtime configuration for execution
+    options = set_default_run_config(options)
+
+    # Compile the kernel with the schedule
+    gemm_prefetch = wave_compile(options, gemm_prefetch, prefetch_schedule)
+
+    # Create test data
+    a = torch.randn(shape[0], shape[2], dtype=torch.float16, device="cuda")
+    b = torch.randn(shape[1], shape[2], dtype=torch.float16, device="cuda")
+    c = torch.zeros(shape[0], shape[1], dtype=torch.float32, device="cuda")
+
+    # Run the kernel
+    gemm_prefetch(a, b, c)
+
+    if is_debug:
+        with open("gemm_prefetch.mlir", "w") as f:
+            f.write(gemm_prefetch.asm)
+
+    expected = torch.matmul(a, b.t()).to(torch.float32)
+    assert torch.allclose(c, expected, rtol=1e-2, atol=1e-2)
+
+    print("GEMM prefetch test passed!")
+
+
+def gemm_prefetch_reorder(is_debug=False):
```

Comment:
I would suggest probably splitting this up into 6.1 6.2 etc. Where the first subsection introduces you to pipelining and the second one introduces to pipe planning along with ping-pong